WordPress/gutenberg
# Env: Add `exec` Command

## What?

_Note: This lacks appropriate documentation updates. Assuming we're fine with the naming and configuration I will update that too._

This pull request adds a new `exec` command that allows you to run configured scripts within the context of a container. You are able to set these at the environment-level and declare a `cwd` that the command should be ran in.

## Why?

Right now to run scripts in your container, you need `package.json` scripts with a verbose syntax. This includes referencing container paths and having to deal with the current working directory. By moving these scripts out of `package.json` we can keep all of the paths and mapping to the environment that actually utilizes them. This means that we can then run `wp-env exec tests-wordpress test:php` and it can run a PHPUnit local to the package.

## How?

We add a new `exec` command and configuration options for `scripts`. This uses the same underlying functionality as `run` but moves the script definition into the `.wp-env.json` file. For example, this would change the following `package.json` script:

`"test:unit": "wp-env run tests-wordpress "/var/www/html/wp-content/plugins/woocommerce/vendor/bin/phpunit -c /var/www/html/wp-content/plugins/woocommerce/phpunit.xml"`

Into the much less verbose:

`"test:unit": "wp-env exec tests test:unit"`

There is no longer any ambiguity about what these paths are referencing or why it's configured this way. Since the container is the place where the script is actually being executed, having it there feels like a convenience.

## Testing Instructions

1. Add the following to your `.wp-env.override.json`. 
```json
{
    "scripts": {
        "test": {
            "script": "ls -la",
            "cwd": "wp-content/plugins/gutenberg"
        }
    }
}
```
2. Run `npm run wp-env -- exec tests test` and note that it prints the directory. 
3. Play around with the `"cwd"` option, it supports absolute paths as well as being removed altogether.
4. Play around with `env.tests` and `env.development` overrides for `"scripts"`.
5. Play around with `run` command to make sure it still works after the refactor.
